server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -1077,7 +1074,8 @@ public void invariant() {
             assert followersChecker.getFastResponseState().term() == getCurrentTerm() : followersChecker.getFastResponseState();
             assert followersChecker.getFastResponseState().mode() == getMode() : followersChecker.getFastResponseState();
             assert (applierState.nodes().getMasterNodeId() == null) == applierState.blocks().hasGlobalBlockWithId(NO_MASTER_BLOCK_ID);
-            assert preVoteCollector.getPreVoteResponse().equals(getPreVoteResponse()) : preVoteCollector + " vs " + getPreVoteResponse();
+            assert REGISTER_MODE_ASSERTIONS || preVoteCollector.getPreVoteResponse().equals(getPreVoteResponse())
```

Comment:
Maybe we should update the pre-vote response in the `PreVoteCollector` after we refresh the accepted state?

Reply:
Yes I'd prefer to keep this assertion in place if possible. `PreVoteCollector#state` is used to respond to pre-vote requests without acquiring a lock so it's kind of important for it to be reasonably up-to-date even though it's not essential for consistency.

Updating the state read during an election should be basically the same as calling `Coordinator#handlePublishRequest`, which does update the `PreVoteCollector` (amongst other things). Maybe we should do that?